### ISSUE TITLE
BUG(gate): Early Access Gate Should Check Network

### ISSUE THREAD

**Issue by alfl** [kubelt/kubelt#651]
## I'm submitting a ...

- [x] bug report
- [x] feature request

## What is the current behavior?

Current behaviour is that the early access gate assumes Ethereum mainnet. If your wallet is set to a different chain, or a different network, the flow continues.

## What is the expected behavior?

Expected behaviour is that the user is warned that they should be on Ethereum mainnet to continue.

## What is the motivation / use case for changing the behavior?

It's not always obvious to users when they are on a different chain and it causes confusion, bad UX, and a potential security vulnerability (creates a rare hash collision problem across compatible chains).

**Comment by alfl** [comment]
Assigned to @picassobanana because we need design for a notice that the user isn't on ETH mainnet.

Overall low priority for now -- other ticketed work should be completed first.

**Comment by alfl** [comment]
https://user-images.githubusercontent.com/9158102/182691910-575daf9d-c723-4e2e-b2c6-8875822872fb.mp4

First run through this shows a prompt to change networks.

**Comment by alfl** [comment]
Not an issue.
